### PR TITLE
Work branch

### PR DESCRIPTION
**Pull request by cioro**
Let's look at what I have done.

### CODE REVIEW

File: etr_definitions.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#ifndef DEFINITIONS_H
```
Comment (thorpe-dev) [inline etr_definitions.h]: I would change this to be `#ifndef INCLUDED_ETR_DEFINITIONS`

File: etr_definitions.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+#define DEFINITIONS
```
Comment (thorpe-dev) [inline etr_definitions.h]: see above

File: etr_report.c [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+//Internal function declarations
+//-------------------------------------------------------
+bool alphcmp (struct executed_trade L, struct executed_trade R);
+void merge(struct executed_trade arr[], int l, int m, int r);
+void mergeSort(struct executed_trade arr[], int l, int r);
+//-------------------------------------------------------
+
+
+void printReport(FILE * freport, struct executed_trade BUYS[], struct executed_trade SELLS[],int NBuys,int NSells)
```
Comment (thorpe-dev) [inline etr_report.c]: I would change this prototype to be:

``` c
void printReport(FILE * freport,
                           struct executed_trade * buyTrades,
                           size_t buysSize,
                           struct executed_trade * sellTrades,
                           size_t sellSize)
```

File: etr_report.c [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+{
+  mergeSort(BUYS,0, NBuys-1);
+   
+  fprintf(freport,"\n");
+  fprintf(freport,"Trade Type: BUY \n");
+  fprintf(freport,"------------------------------------------ \n");
+  for(int i = 0; i < NBuys; ++i){
```
Comment (thorpe-dev) [inline etr_report.c]: `size_t`, not `int`

File: etr_report.c [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+   
+   
+    if(i!=0 && strcmp(BUYS[i-1].stock_name,BUYS[i].stock_name)){
```
Comment (thorpe-dev) [inline etr_report.c]: format this to be:

``` c
if (i != 0 && strcmp(buyTrades[i-1].stock_name, buyTrades[i].stock_name))
{
```

File: etr_report.c [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+      char name[]="User";
+      char price[]="Price";
+      char shares[]="Shares";
+      fprintf(freport,"%-8s \t %-8s \t %-8s \n",name,price,shares);
+    }
+    fprintf(freport,"%-8s \t %-.2f \t %-2i\n",SELLS[i].user_name,SELLS[i].traded_price,SELLS[i].num_shares);
+  }
+}
+
+bool alphcmp (struct executed_trade L, struct executed_trade R)
```
Comment (thorpe-dev) [inline etr_report.c]: better names here, pass by pointer not by value

File: etrstock.h [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+struct Node_stock{
+  struct stock data;
+  struct Node_stock * next;
+};
+
+struct Node_stock * createStockList(FILE * file, struct Node_stock * head );
+struct Node_stock *listSort(struct Node_stock *list);
+double look_up_stock_price(char search_name[MAX_STOCK_NAME_LENGTH],struct Node_stock* head);
```
Comment (thorpe-dev) [inline etrstock.h]: `char *`

File: etrtrade.c [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+   }
+  tmp2[j] = '\0';
+  
+  t.user_price = atof(tmp2);
+  
+  return t;
+}
+
+
+void process_trade(struct executed_trade BUYS[], struct executed_trade SELLS[], struct trade_request t,double stock_price,int* buys, int* sells){
```
Comment (thorpe-dev) [inline etrtrade.c]: split this up into multiple lines

File: m.c [hunk 9]
```diff
@@ -0,0 +1,52 @@
+//This file test the behaviour of all the functions and structs involved in
```
Comment (thorpe-dev) [inline m.c]: I would name this file `etr.m.c` or similar

File: etrtrade.c (hunk 8 above)
Comment (cioro) [inline etrtrade.c]: Should I make struct executed_trade into a pointer? 
void process_trade(struct executed_trade \* BUYS, 
                                struct executed_trade \* SELLS, 
                                struct trade_request t,
                                double stock_price,
                                int\* buys, 
                                int\* sells)

**Comment by cioro** [comment]
I updated the branch with your suggestions. Any more suggestions? Thank you for your feedback!

File: etr_report.c [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+  }
+ 
+  mergeSort(sellTrades,0, sellSize-1);
+  fprintf(freport,"\n");
+  fprintf(freport,"Trade Type: SELL \n");
+  fprintf(freport,"------------------------------------------ \n");
+  for(size_t i = 0; i < sellSize; ++i){
+   
+   
+    if(i!=0 && strcmp(sellTrades[i-1].stock_name,sellTrades[i].stock_name)){
```
Comment (thorpe-dev) [inline etr_report.c:52]: You need to fix the formatting here (and elsewhere)

File: etr_report.c [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+      char name[]="User";
+      char price[]="Price";
+      char shares[]="Shares";
+      fprintf(freport,"%-8s \t %-8s \t %-8s \n",name,price,shares);
+    }
+    fprintf(freport,"%-8s \t %-.2f \t %-2i\n",sellTrades[i].user_name,sellTrades[i].traded_price,sellTrades[i].num_shares);
+  }
+}
+
+bool alphcmp (const struct executed_trade* left_elem, const struct executed_trade* right_elem)
```
Comment (thorpe-dev) [inline etr_report.c]: I think this method needs some comments + description saying what it does

File: etrtrade.c [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+   }
+  tmp2[j] = '\0';
+  
+  t.user_price = atof(tmp2);
+  
+  return t;
+}
+
+
+void process_trade(struct executed_trade BUYS[],
```
Comment (thorpe-dev) [inline etrtrade.c]: same comments as before here - all CAPS is a bad naming method for variables

File: etr_report.c (hunk 10 above)
Comment (cioro) [inline etr_report.c:52]: It is not obvious to me which are the formatting issues?

**Pull request merged**